python/cuml/benchmark/bench_helper_funcs.py

```diff
@@ -26,3 +26,40 @@ def fit(m, x, y=None):
 
 def fit_transform(m, x):
     m.fit_transform(x)
+
+
+def predict(m, x, y=None):
+    m.predict(x) if y is None else m.predict(x, y)
+
+
+def _fil_classification_setup(m, data, arg={}):
+    from cuml.utils.import_utils import has_xgboost
+    if has_xgboost():
+        import xgboost as xgb
+    else:
+        raise ImportError("No XGBoost package found which is required for benchmarking FIL")
+    import os 
+
+    tmp_path = "./fil_models"
+    if not os.path.exists(tmp_path):
+        os.makedirs(tmp_path)
+
+    # use maximum 1e6 rows to train the model 
+    train_size = min(data[0].shape[0], 1000000)
```

Comment:
Do we need so much data for a test run? Is the concern that we won't get adequate depth with a smaller dataset? Seems like it would be nice if we could get by with like 100k rows to speed stuff up, but I honestly don't know if that will change the tree structure enough at useful depths to make a difference.

Reply:
1e6 is the maximum number of samples for training the model as suggested by @canonizer

Reply:
Yes, I'd prefer to have a larger dataset for training and testing, at least 1e6 for either. @JohnZed Speaking of speeding up: how often are we planning to run these benchmarks? Also, are we planning to run them as part of the CI?

Reply:
Ops is adding nightly runs to start but hoping to add them as an "advisory" check on each CI change (so you'd get a score, but it would not cause your change to fail). So if tests get expensive, it will bog things down. I'm ok deferring some optimizations until closer to that point though.

Reply:
If it gets expensive, feel free to reduce the size of the training dataset. 

Also, independent of the above, it makes sense to print forest statistics in this case, e.g. the average depth and average number of nodes in the trees. This is for a different pull request, though.

Reply:
So maybe we should change 1e6 to 1e5?